pkg/maybe/maybe.go

```diff
@@ -0,0 +1,28 @@
+package types
```

Comment:
I want to us to make sure to understand if we really need this or if a an error pointer would suffice.

Alternatively, this is really reminding me of optional. There's no native implementation in go, but packages are available: https://github.com/markphelps/optional

Reply:
I don't fell like it's a good solution for our use case.

`optional` package seem to do more thinks like serialization which we don't need in our case.

Also this Maybe type is actually misnamed. It does not represent the presence or absence of a value but rather the presence of a value or an error which translates in functional programming to the `Either` type. The closest external implementation would be this one https://github.com/olehmushka/golang-toolkit/blob/main/either/either.go

Will update it to reflect this.

Reply:
`Either` makes it clearer. Will re-review accordingly.